docs/reference/di/providing-different-implementation-of-a-service.md

```diff
@@ -56,7 +56,7 @@ object MainApp extends ZIOAppDefault {
       _        <- ZIO.debug(s"Application started with $poolSize pool size.")
     } yield ()
 
--  def run = myApp.provideLayer(AppConfig.appArgsLayer)
+-  def run = myApp.provideSome[ZIOAppArgs](AppConfig.appArgsLayer)
```

Comment:
I would use `provide` and `provideSome` consistently here.

Reply:
That is exactly what I intended here, or did I get you wrong?

We use `myApp.provideSome[ZIOAppArgs](AppConfig.appArgsLayer)` in the code block above. That's why I changed the diff here as well. So it explains to a reader that we now remove the configuration from app args and use system properties instead.

Remove: `myApp.provideSome[ZIOAppArgs](AppConfig.appArgsLayer)`
Insert: `myApp.provideLayer(AppConfig.systemEnvLayer)`

This is how it would be rendered:

<img width="714" alt="Screenshot 2023-03-17 at 10 12 11" src="https://user-images.githubusercontent.com/7292949/225862097-d6786e00-5b0a-4a6f-94f2-a4258d818b72.png">

Reply:
Right. if we were using `provideSome` before we should use `provide` now.

Basically we have two "axes". One is whether we are using manual layer construction (`provideLayer` or `provideSomeLayer`) versus automatic layer construction (`provide` or `provideSome`). The second is whether we are providing all of the environment (`provide` or `provideLayer`) or some of it (`provideSome` or `provideSomeLayer`).

So these lines are somewhat confusing right now if that we are trying to illustrate how we use a different operator when we have a layer that doesn't have any other requirements but we are also changing back and forth between automatic and manual layer construction.

Reply:
Ah, I didn't notice! 

So let's say we use automatic layer construction for this example (since all other examples on that page use it, too). Then we would have the following lines:

- `myApp.provideSome[ZIOAppArgs](AppConfig.appArgsLayer)`
- `myApp.provide(AppConfig.systemEnvLayer)`

For the configuration via arguments, we use `provideSome` since we don't provide `ZIOAppArgs` yet. For the configuration via system properties, we use `provide`, since we completely construct the layer.

WDYT?

Reply:
Exactly!

Reply:
👍 I updated the commit.